### ISSUE TITLE
votable.converters.NumericArray().parse() bug

### ISSUE THREAD

**Issue by al-niessner** [astropy/astropy#4781]
Trying to read VOTables returned from the Gemini Telescope Archive. Tracked it down to erroneous code in votable/converters.py.

**Comment by astrofrog** [comment]
@al-niessner - please provide an example of a file that causes the error and also report the actual error here, otherwise we will not be able to fix this.

**Comment by al-niessner** [comment]
I am fixing the code now. Working through the contributing documentation to follow your process.

**Comment by al-niessner** [comment]
In votable/tree.py lines 939 and 975 at least, the call on `votable.converters.NumericArray().parse()` is made without any config parameter allowing it to default to None. The latter was being used with the v1.2 Tables from the Gemini Telescope Archive.

The original code looked like:

```
        if config is None: config = {}
        if config['version_1_3_or_later'] and value == '':
            return np.zeros(self._arraysize, dtype=self._base.format), True
```

Obviously since config was None from the tree.py calls, it was being set to an empty dictionary. The second condition then had to fail because the dictionary is necessarily empty. To preserve all previous logic, I simply combined the two if statements into a singular elif:

```
        if config is None: config = {}
        elif config['version_1_3_or_later'] and value == '':
            return np.zeros(self._arraysize, dtype=self._base.format), True
```

Such change should behave identically with calls that provide a config and the condition must necessarily be false when no config is provided making it sufficient to use the ifel construct.

It was tested with the gemini case fixing that problem.

**Comment by al-niessner** [comment]
Fix has been submitted on pull request 4872.

**Comment by pllim** [comment]
Fixed by #4782.
